### PR TITLE
switching AvailabilityZone from AllowedValues to AllowedPatternRegex

### PR DESCRIPTION
**Pull request by PatMyron**
fixes https://github.com/aws-cloudformation/cfn-lint/issues/1966
https://aws.amazon.com/blogs/aws/now-open-third-availability-zone-in-the-aws-china-beijing-region/

---

https://aws.amazon.com/blogs/aws/category/regions/
https://aws.amazon.com/about-aws/global-infrastructure/
https://aws.amazon.com/about-aws/global-infrastructure/localzones/locations/
https://aws.amazon.com/wavelength/features/#Locations

### CODE REVIEW

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types.json [hunk 1]
```diff
@@ -4,33 +4,7 @@
...
-          "me-south-1a", "me-south-1b", "me-south-1c",
-          "sa-east-1a", "sa-east-1b", "sa-east-1c",
-          "us-east-1a", "us-east-1b", "us-east-1c", "us-east-1d", "us-east-1e", "us-east-1f", "us-east-1-bos-1a", "us-east-1-iah-1a", "us-east-1-mia-1a",
-          "us-east-2a", "us-east-2b", "us-east-2c",
-          "us-gov-east-1a", "us-gov-east-1b", "us-gov-east-1c",
-          "us-gov-west-1a", "us-gov-west-1b", "us-gov-west-1c",
-          "us-west-1a", "us-west-1b", "us-west-1c",
-          "us-west-2a", "us-west-2b", "us-west-2c", "us-west-2d", "us-west-2-lax-1a", "us-west-2-lax-1b"
-        ],
+        "AllowedPatternRegex": "[a-z0-9-]+",
```
Comment (kddejong) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types.json:7]: I get the value in this I just wonder if there is a better regex for this? 
`^[a-z]{2}-[a-z0-9-]+-[0-9][a-z]$`

Comment (PatMyron) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types.json:7]: open to suggestions, that specific pattern would miss [wavelength zones](https://aws.amazon.com/wavelength/features/#Locations) though

Comment (PatMyron) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types.json:7]: Was considering these two stricter patterns as well:
`[a-z]{2}-[a-z0-9-]+`
`(af|ap|ca|cn|eu|me|sa|us)-[a-z0-9-]+`
but wasn't sure if those patterns might eventually change

Comment (kddejong) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types.json:7]: Lets do what you have we can always improve it later.

**Pull request merged**